code/modules/vehicles/tank.dm

```diff
@@ -0,0 +1,381 @@
+	//Who's driving the tank
+	var/mob/living/carbon/human/pilot
+	var/mob/living/carbon/human/gunner
+	var/list/operators = list() //Who's in this tank? Prevents you from entering the tank again
+	//Health and combat shit
+	var/obj/turret_overlay/turret_overlay //Allows for independantly swivelling guns, wow!
+	var/obj/turret_overlay/minigun/minigun_overlay
+	var/obj/item/tank_weapon/main_cannon //What we use to shoot big shells
+	var/obj/item/tank_weapon/minigun/minigun //What we use to shoot mini shells ((rapidfire xenocrusher 6000))
+	var/main_cannon_dir = null //So that the guns swivel independantly
+	var/minigun_dir = null
+	var/atom/firing_target = null //Shooting code, at whom are we firing?
+	var/firing_main_cannon = FALSE
+	var/firing_minigun = FALSE
+	var/last_drive_sound = 0 //Engine noises.
+
+/obj/vehicle/tank/examine(mob/user)
+	. = ..()
+	to_chat(user, "<b>To fire its main cannon, <i>ctrl</i> click a tile</b>")
+	to_chat(user, "<b>To fire its minigun, click a tile</b>")
+
+/obj/turret_overlay
+	name = "Tank gun turret"
+	desc = "The shooty bit on a tank."
+	icon = 'icons/obj/tank_gun.dmi'
+	icon_state = "turret"
+	layer = ABOVE_OBJ_LAYER
+	animate_movement = TRUE //So it doesnt just ping back and forth and look all stupid
+	mouse_opacity = FALSE //It's an overlay
+
+/obj/vehicle/tank/Initialize()
+	. = ..()
+	turret_overlay = new()
+	update_icon()
+	main_cannon = new(src) //Make our guns
+	minigun = new(src)
+	main_cannon.owner = src
+	minigun.owner = src
+	GLOB.tank_list += src
+
+/obj/vehicle/tank/Destroy()
+	qdel(turret_overlay)
+	qdel(main_cannon)
+	qdel(minigun)
+	. = ..()
+
+/obj/vehicle/tank/Move()
+	. = ..()
+	update_icon()
+	if(world.time > last_drive_sound + 2 SECONDS)
+		playsound(src, 'sound/ambience/tank_driving.ogg', vol = 20, sound_range = 30)
+		last_drive_sound = world.time
+
+/obj/vehicle/tank/update_icon() //To show damage, gun firing, whatever. We need to re apply the gun turret overlay.
+	var/icon/I = icon(icon,icon_state,dir)
+	bound_width = I.Width() //Adjust the hitbox in case admins want to make an OMEGAtank
+	bound_height = I.Height()
+	vis_contents = null
+	if(!turret_overlay) //We have to do this because of BYOND's innate dir setting habits, in byond, if you have an overlay'd object its direction can ONLY be which way the source is facing. So, we improvise.. ((yes this is kinda shit))
+		return
+	if(main_cannon_dir)
+		turret_overlay.setDir(main_cannon_dir)
+		if(main_cannon_dir == WEST || main_cannon_dir == SOUTHWEST|| main_cannon_dir == NORTHWEST)
+			turret_overlay.pixel_x = pixel_x -19
+		else
+			turret_overlay.pixel_x = 0
+	vis_contents += turret_overlay
+
+/obj/vehicle/tank/attack_hand(mob/user)
+	if(pilot && gunner)
+		to_chat(user, "You are unable to enter [src] because all of its seats are occupied!")
+		return ..()
+	if(user in operators)
+		exit_tank(user)
+		return
+	var/position = alert("What seat would you like to enter?.",name,"pilot","gunner") //God I wish I had radials to work with
```

Comment:
`Provide options based on what's available, though you'll still need to check the position wasn't filled since the options were shown to the player`

Reply:
BYOND won't let you do inputs of 1 thing, it just automatically makes you choose the one option.